### PR TITLE
Global Auth not supported by Client

### PR DESCRIPTION
**Pull request by GregHanson**
Documentation suggests the Client can be created with an empty AuthToken to support Global auth mode

```
// Config stores the configurable attributes of the client.
type Config struct {

	// URL of the registry server.
	URL string

	// AuthToken is the bearer token to be used for authentication with the registry.
	// If left empty, no authentication is used.
	AuthToken string
```

but `normalizeConfig()` complains when an AuthToken is not provided.  Is this check no longer valid?  Or should there be another field in the `Config` object for AuthType (global/trusted/jwt) ?  Then the error can  be thrown when AuthToken is empty and AuthType is not `global`

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -279,10 +279,5 @@ func normalizeConfig(config *Config) error {
 		return newError(ErrorCodeInvalidConfiguration, fmt.Sprintf("unsupported scheme %s", url.Scheme), nil, "")
 	}
 
-	url.Query()
```
Comment (GregHanson) [inline client/client.go]: I noticed `.Query()` is called and never used - is this an oversight or remnant of previous logic?

Comment (rshriram) [inline client/client.go]: @GregHanson is this line change related to the bug described by the issue?

Comment (zcahana) [inline client/client.go]: @GregHanson That's indeed an oversight and should be removed. Thanks for spotting.
@rshriram it's only spatially related, but can definitely go in along with the token validation fix.

**Comment by zcahana** [comment]
@GregHanson As you pointed out, the token validation is no longer valid - a remnant of pre-global auth days. So, LGTM --> merging.
